src/databricks/labs/remorph/snow/snowflake.py

```diff
@@ -228,6 +228,12 @@ class Tokenizer(Snowflake.Tokenizer):
             r"(?i)var\s+\w+\s+=\s+\w+?": TokenType.VAR,
         }
 
+        SINGLE_TOKENS = {
+            **Snowflake.Tokenizer.SINGLE_TOKENS,
+            "&": TokenType.PARAMETER,  # https://docs.snowflake.com/en/user-guide/snowsql-use#substituting-variables-in-a-session
+            "!": TokenType.COMMAND,
```

Comment:
The program behaves as follows:
1)If the SnowSQL command starts with an exclamation mark !, it will generate the output as commented with --.
Example:

Input: !(2 = 2)
Output: -- !'(2 = 2)'

synatx tree: Command(
  this=!,
  expression=Literal(this=(2 = 2), is_string=True))

2)If the exclamation mark ! appears within the query but not at the starting position, it will generate the equivalent DBSQL output.
Ex:
input: select * from emp where !(2 = 2)
output: SELECT * FROM emp WHERE  !(2 = 2) 
syntax tree: Select(
  expressions=[
    Star()],
  from=From(
    this=Table(
      this=Identifier(this=emp, quoted=False))),
  where=Where(
    this=Anonymous(
      this=!,
      expressions=[
        EQ(
          this=Literal(this=2, is_string=False),
          expression=Literal(this=2, is_string=False))])))

Reply:
add a unit test to prove it with automation.

```sql
!whatever
SELECT !(2 = 2) AS always_false
```

and make sure it translates.

Reply:
added the test case